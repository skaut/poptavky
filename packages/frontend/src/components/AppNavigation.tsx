import type React from "react";

import { Navigation } from "./Navigation";

export const AppNavigation = (): React.JSX.Element => (
  <Navigation
    items={[
      {
        link: "/",
        title: "PoptÃ¡vky",
      },
      {
        link: "/projekty",
        title: "Projekty",
      },
    ]}
  />
);
