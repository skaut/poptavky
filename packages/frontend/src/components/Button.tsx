import React from "react"
import styled from "@emotion/styled"
import { theme } from "../theme"
import { ExtLink, LinkProps } from "./Link"

interface Props extends LinkProps {}

export const Button: React.FC<Props> = (props) => <ButtonWrapper {...props} />

const ButtonWrapper = styled(ExtLink)`
  display: inline-block;
  font-size: 1.2rem;
  background-color: ${theme.colors.brand};
  color: white;
  margin-top: 0.2rem;
  padding: 0.6rem 1.2rem;
  border-radius: 10px;
  box-shadow: 0 0 0 ${theme.colors.gray};
  transition: box-shadow 300ms;
`
