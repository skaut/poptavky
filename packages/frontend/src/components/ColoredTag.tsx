import { css } from "@emotion/react";
import styled from "@emotion/styled";
import stringToColor from "string-to-color";

export const ColoredTag = styled("span")(
  (props: { children: string; isLight?: boolean }) => css`
    background-color: ${stringToColor(props.children)}33;
    color: ${props.isLight === true ? "white" : "#444;"};
    font-family: themix;
    font-size: 0.8rem;
    margin-right: 8px;
    padding: 0.2rem 0.4rem;
    border-radius: 4px;
    ${props.isLight === true ? "border: 1px solid #ffffff55" : ""};
  `,
);
