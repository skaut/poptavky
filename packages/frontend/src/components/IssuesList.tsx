/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import type React from "react";
import ReactMarkdown from "react-markdown";
import { Link } from "react-router-dom";
import remarkGfm from "remark-gfm";

import type { Project } from "../interfaces/Project";
import type { ProjectInfo } from "../interfaces/ProjectInfo";
import type { ProjectIssue } from "../interfaces/ProjectIssue";
import type { ProjectListing } from "../interfaces/ProjectListing";
import { theme } from "../theme";
import { getIssueLink } from "../utils/getIssueLink";
import { H3 } from "./Typography";

export const IssuesList = ({
  issues,
  project,
}: {
  readonly issues: Array<ProjectIssue>;
  readonly project: ProjectListing | (Project & ProjectInfo);
}): React.JSX.Element => (
  <>
    {issues.map((issue) => (
      <article
        css={css`
          margin: 12px 0 24px;
          img {
            max-width: 100%;
          }
        `}
        key={issue.number}
      >
        <H3>
          <Link
            to={getIssueLink({
              project: {
                ...("info" in project
                  ? { ...project, ...project.info }
                  : { ...project }),
              },
              ...issue,
            })}
          >
            {issue.title}
          </Link>
          &nbsp;
          <span
            css={css`
              color: ${theme.colors.gray};
              font-weight: normal;
            `}
          >
            #{issue.number}
          </span>
        </H3>
        <ReactMarkdown remarkPlugins={[remarkGfm]}>
          {issue.description}
        </ReactMarkdown>
      </article>
    ))}
  </>
);
