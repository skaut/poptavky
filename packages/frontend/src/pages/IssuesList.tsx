/** @jsxImportSource @emotion/react */
import type React from "react"

import { Issue } from "../components/Issue"
import type { ProjectListings } from "../interfaces/ProjectListings"
import { getIssuesWithProjectInfo } from "../utils/getAllIssues"

export const IssuesList = ({
  data,
}: {
  readonly data: ProjectListings
}): React.JSX.Element => {
  const issues = getIssuesWithProjectInfo(data)
  return (
    <>
      {issues.map((issue) => (
        <Issue issue={issue} key={issue.link} />
      ))}
    </>
  )
}
