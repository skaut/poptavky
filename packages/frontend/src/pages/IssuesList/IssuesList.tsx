/** @jsxImportSource @emotion/react */
import { getIssuesWithProjectInfo } from "../../utils/getAllIssues"
import { Issue } from "../../components/Issue"
import { ProjectListings } from "../../interfaces/ProjectListings"

export const IssuesList = ({ data }: { data: ProjectListings }) => {
  const issues = getIssuesWithProjectInfo(data)
  return (
    <>
      {issues.map((issue) => (
        <Issue key={issue.link} issue={issue} />
      ))}
    </>
  )
}
