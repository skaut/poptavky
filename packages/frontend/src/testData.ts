import { ProjectListings } from "./interfaces/ProjectListings"

export const testData: ProjectListings = {
  projects: [
    {
      owner: "skaut",
      repo: "skaut-google-drive-gallery",
      info: {
        name: "WordPress plugin: Galerie obrázků a videí z Google Disku",
        "short-description": "Plugin pro WordPress, který ze složek s fotkami na Google Disku dělá galerie na webu.",
        description: "Galerie obrázků a videí z Google Disku je plugin pro WordPress, který propojí web s Google Diskem. V pluginu se dá vybrat libovolná složku na Google Disku nebo ve Sdíleném disku a plugin z ní vytvoří galerii s obrázky a videi na libovolné stránce či příspěvku na webu. Ze všech složek a podsložek jsou navíc vytvořeny vnořené “podgalerie”. Plugin vzniknul primárně pro potřeby Junáka, oddílů a středisek, ale vyvíjíme ho tzv. Open-Source (otevřeně pro veřejnost) a nabízíme ho v oficiální databázi pluginů pro WordPress. V tuto chvíli (léto 2021) podle statistik WordPressu plugin běží na víc jak 4 000 webových stránek a z kontaktů s uživateli víme, že ho využívají jak oddíly a střediska v Junáku, tak i \"náhodná veřejnost\" - zvlášť oblíbený se zdá být u skautů z celého světa a profesionálních fotografů, prezentujících svoje fotky.",
        tags: [
          "wordpress",
          "php",
          "javascript",
          "google"
        ],
        maintainers: [
          {
            name: "Marek Dědič – Mlha",
            email: "mlha@skaut.cz"
          }
        ],
        links: [
          {
            type: "email",
            uri: "mailto:mlha@skaut.cz"
          },
          {
            type: "homepage",
            uri: "https://wordpress.org/plugins/skaut-google-drive-gallery/"
          },
          {
            type: "demo",
            uri: "https://demo-skaut-google-drive-gallery.skauting.cz/"
          },
          {
            type: "issue-tracker",
            uri: "https://github.com/skaut/skaut-google-drive-gallery/issues"
          },
          {
            type: "docs",
            uri: "https://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery"
          },
          {
            type: "github-repo",
            uri: "https://github.com/skaut/skaut-google-drive-gallery",
            name: "skaut/skaut-google-drive-gallery"
          }
        ]
      },
      issues: [
        {
          number: 1,
          title: "Add hint to set videos to public",
          description: "When editing and selecting a folder with videos, the editor could display a notice that setting videos to \"Anyone can view\" will speed them up.\r\n\r\nDepends on #792",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/873"
        },
        {
          number: 2,
          title: "If a shortcut name differs from the original, the name in gallery and breadcrumbs differs",
          description: "When in a situation that there is a Google Drive shortcut somewhere in the gallery, the plugin displays the folder and it works for browsing. However, when looking at the shortcut as a member of its parent directory and when being inside the directory and looking at its name in the breadcrumbs, one uses the shortcut name, whereas the other uses the original folder name (when they differ).\r\n\r\nThis is a bug and both should show the shortcut name.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/745"
        },
        {
          number: 3,
          title: "Editing doesn't check path validity",
          description: "When opening the editor which contains a gallery that is made from a folder that has been renamed or deleted, the editor breaks. It shows: `Directory &quot;Galerie ODYWEB&quot; wasn&#039;t found - it may have been deleted or renamed.` Instead, it should verify that the path is valid and if it's not, should display a warning (and possibly move you to either the root directory, or the closest valid ancestor - however, this should not lead to you accidentally changing the gallery!)\r\n\r\nLast tested on 36dc1a10a353ba5bf2b98f33b49b1135a36ae3d3",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/683"
        },
        {
          number: 4,
          title: "Ctrl-click on folders does not work",
          description: "Ctrl+click on a gallery item (subfolder) or breadcrumbs should open in a new tab",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/571"
        },
        {
          number: 5,
          title: "Add option to specify gallery by a Drive link",
          description: "https://wordpress.org/support/topic/folder-link-instead-of-folder-name/\r\n\r\nBasically, instead of searching for the folder in WP, you'd just paste the link to it into the editor and the plugin would figure out the path (Google Drive files now return a list of parents, so it could be done.)\r\n\r\nDepends on #111.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/311"
        },
        {
          number: 6,
          title: "Add images to the block",
          description: "Currently, images aren't listed in the editor - both block and TinyMCE plugin. This often confuses users.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/257"
        },
        {
          number: 7,
          title: "Add video overlay",
          description: "In the image grid, we should add an overlay to signify that an item is a video (for example a big play icon).",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/235"
        },
        {
          number: 8,
          title: "Increase test coverage",
          description: "https://codecov.io/gh/skaut/skaut-google-drive-gallery",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/164"
        },
        {
          number: 9,
          title: "Themes with \"editor-styles\" support change the look of the block editor",
          description: "If the active theme contains\r\n```php\r\nadd_theme_support( 'editor-styles' );\r\n```\r\nits styles are also applied to the block editor, which then looks weird. We should implement our own styles for the editor.\r\n\r\nExample of such theme is the Twenty Seventeen theme.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/98"
        },
        {
          number: 10,
          title: "Google app creation manual is outdated",
          description: "https://napoveda.skaut.cz/dobryweb/en-skaut-google-drive-gallery/en-get-google-application\r\nhttps://napoveda.skaut.cz/dobryweb/cs-skaut-google-drive-gallery/cs-ziskani-google-aplikace\r\n\r\nGoogle changed the consent screen layout and require some form of verification.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/92"
        },
        {
          number: 11,
          title: "Directory preview images",
          description: "Currently, the first image in a directory is used for its preview.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/54"
        },
        {
          number: 12,
          title: "UX & design audit",
          description: "Go over the design of:\r\n- [ ] gallery view https://github.com/skaut/skaut-google-drive-gallery/issues/36\r\n- [ ] preview\r\n- [ ] folder thumbnail - overlay, folder title and image count\r\n- [ ] svg folder icon\r\n- [ ] gallery breadcrumbs\r\n- [ ] options page\r\n- [ ] WP.org plugin page\r\n- [ ] TinyMCE & Gutenberg editor plugins\r\n- [ ] TinyMCE & Gutenberg \"Authentication failed\" dialogs\r\n\r\nEverything on the frontend needs to work at least with the default WP template and DSW templates.",
          link: "https://github.com/skaut/skaut-google-drive-gallery/issues/33"
        }
      ]
    },
    {
      owner: "skaut",
      repo: "skautis-integration",
      info: {
        name: "WordPress plugin: SkautIS integrace",
        "short-description": "Plugin pro WordPress, který napojuje přihlašování přes skautIS na skautské weby.",
        description: "Plugin zajišťuje propojení skautských webů se skautISem. Dá se přes plugin registrovat, přihlašovat. Dokáže u stránek nastavovat oprávnění podle roli a dalších nastavení ve skautISu. Skrývá obsah podle toho jaká pravidla na stránce jsou nastavená.",
        tags: [
          "wordpress",
          "php",
          "javascript",
          "skautis"
        ],
        maintainers: [
          {
            name: "Michal Janata – Kalich",
            email: "kalich@skaut.cz"
          },
          {
            name: "Marek Dědič – Mlha",
            email: "mlha@skaut.cz"
          }
        ],
        links: [
          {
            type: "email",
            uri: "mailto:kalich@skaut.cz"
          },
          {
            type: "homepage",
            uri: "https://wordpress.org/plugins/skautis-integration/"
          },
          {
            type: "issue-tracker",
            uri: "https://github.com/skaut/skautis-integration/issues"
          },
          {
            type: "docs",
            uri: "https://napoveda.skaut.cz/skautis/skautis-integration"
          },
          {
            type: "github-repo",
            uri: "https://github.com/skaut/skautis-integration",
            name: "skaut/skautis-integration"
          }
        ]
      },
      issues: [
        {
          number: 1,
          title: "User management smart table is not smart when the registration module is enabled",
          description: "",
          link: "https://github.com/skaut/skautis-integration/issues/115"
        },
        {
          number: 2,
          title: "Chyba v zobrazení",
          description: "![image](https://user-images.githubusercontent.com/8988898/130583907-9aaf6288-2129-42dd-88ef-0a27b0c7ddd1.png)\r\n",
          link: "https://github.com/skaut/skautis-integration/issues/109"
        },
        {
          number: 3,
          title: "WordPress 5.x + Gutenberg",
          description: "Musíme pořešit, jak nejlépe integrovat náš plugin s WP 5.x a hlavně Gutenbergem.\r\n\r\nSkrývání bloků, celých stránek... atd",
          link: "https://github.com/skaut/skautis-integration/issues/70"
        },
        {
          number: 4,
          title: "Profilové fotky",
          description: "Co do WordPressu tahat i profilové fotky, co si lidi do skautISu přidávají. Aby u profilu nějakou fotku měli...",
          link: "https://github.com/skaut/skautis-integration/issues/43"
        },
        {
          number: 5,
          title: "Výběry pravidel v shortcode",
          description: "Přidat Select2 do výběru pravidel v shortcode.",
          link: "https://github.com/skaut/skautis-integration/issues/39"
        },
        {
          number: 6,
          title: "Modul \"Nábory\"",
          description: "modul pro nábory se zobrazením náborové mapy\r\npříklad: http://is.skaut.cz/napoveda/programatori.Hotove-reseni-MapAdvertising-Interaktivni-mapa-naborovych-informaci-zakladni.ashx\r\n\r\nOtevírám k tomu diskuzi...",
          link: "https://github.com/skaut/skautis-integration/issues/25"
        }
      ]
    }
  ]
}