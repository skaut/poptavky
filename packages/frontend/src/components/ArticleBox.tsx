import React from "react"
import ReactMarkdown from "react-markdown"
import styled from "@emotion/styled"
import { ColoredTag } from "./ColoredTag"
import { Link } from "react-router-dom"
import { Article, H2, Paragraph } from "./Typography"
import remarkGfm from "remark-gfm"

export const ArticleBox: React.FC<{
  title: string
  link?: string
  subtitle?: string
  subtitleLink?: string
  subtitleDescription?: string
  description: string
  tags?: string[]
}> = ({
  title,
  link,
  subtitle,
  subtitleLink,
  subtitleDescription,
  description,
  tags,
}) => (
  <ThinArticle>
    <H2>{link ? <Link to={link}>{title}</Link> : title}</H2>
    <ProjectName>
      {subtitle && subtitleLink && (
        <Link to={subtitleLink} title={subtitleDescription}>
          {subtitle}
        </Link>
      )}
    </ProjectName>
    <Paragraph>
      <ReactMarkdown remarkPlugins={[remarkGfm]}>{description}</ReactMarkdown>
    </Paragraph>
    {(tags ?? []).map((tag) => (
      <ColoredTag key={tag}>{tag}</ColoredTag>
    ))}
  </ThinArticle>
)

const ProjectName = styled("p")`
  margin: 0 0 0.6em;
  * {
    font-size: 0.8rem;
  }
`

const ThinArticle = styled(Article)`
  max-width: 600px;
`
