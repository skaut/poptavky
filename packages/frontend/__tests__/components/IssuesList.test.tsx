import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"

import { IssuesList } from "../../src/components/IssuesList"
import { testData } from "../testData"

const project = testData.projects[0]

describe("IssuesList component", () => {
  test("should render correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <IssuesList issues={project.issues} project={project} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("should render correctly without link", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <IssuesList
            issues={project.issues.map((issue) => ({
              ...issue,
              link: undefined,
            }))}
            project={project}
          />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
