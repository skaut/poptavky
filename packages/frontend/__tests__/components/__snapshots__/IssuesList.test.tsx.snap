// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesList component should render correctly 1`] = `
.emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
  class="emotion-0"
>
  <h3
    class="emotion-1"
  >
    <a
      href="/skaut/skaut-google-drive-gallery/1"
    >
      Add hint to set videos to public
    </a>
     
    <span
      class="emotion-2"
    >
      #
      1
    </span>
  </h3>
  <p>
    When editing and selecting a folder with videos, the editor could display a notice that setting videos to "Anyone can view" will speed them up.
  </p>
  

  <p>
    Depends on #792
  </p>
</article>
`;

exports[`IssuesList component should render correctly without link 1`] = `
.emotion-0 {
  margin: 12px 0 24px;
}

.emotion-0 img {
  max-width: 100%;
}

.emotion-1 {
  color: #294885;
  margin: 0 0 12px;
  font-family: themix;
}

.emotion-2 {
  color: #999;
  font-weight: normal;
}

<article
  class="emotion-0"
>
  <h3
    class="emotion-1"
  >
    <a
      href="/skaut/skaut-google-drive-gallery/1"
    >
      Add hint to set videos to public
    </a>
     
    <span
      class="emotion-2"
    >
      #
      1
    </span>
  </h3>
  <p>
    When editing and selecting a folder with videos, the editor could display a notice that setting videos to "Anyone can view" will speed them up.
  </p>
  

  <p>
    Depends on #792
  </p>
</article>
`;
